util/chplenv/chpl_compiler.py

```diff
@@ -3,9 +3,17 @@
 import os
 import sys
 
-from distutils.spawn import find_executable
-
-import chpl_platform, chpl_locale_model, overrides
+try:
+    # Module `distutils` is deprecated in Python 3.10 and will be removed in Python 3.12
+    # Prefer `shutil.which` in Python 3.2+
+    from shutil import which
+except ImportError:
+    # Backport for pre Python 3.2
+    from distutils.spawn import find_executable as which
```

Comment:
The `util/chplenv` scripts are part of our core scripts which still require python 2 support, so I agree we need this here. It would be nice to put this into a `which` wrapper in `util/chplenv/utils.py`, but I'm also happy to do that in a followup PR.